src/shared/directives/conditional-show/domain/conditional-show.model.ts

```diff
@@ -4,21 +4,34 @@ import { FieldsUtils } from '../../../services/fields';
 export class ShowCondition {
```

Comment:
the logic seems ok to me. But this class has grown a lot in complexity over time. I can see that you did some good restructuring, but I think it's still a bit complex. Any way we can further simplify it @ganganna-peddapothu? for example introduce more private methods with descriptive names, or brake it down in multiple classes, etc...